vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -56,36 +79,59 @@ window.Vaadin.Flow.comboBoxConnector = {
 
       const filterChanged = params.filter !== lastFilter;
       if (filterChanged) {
-        pageCallbacks = {};
         cache = {};
         lastFilter = params.filter;
+        this._debouncer = Debouncer.debounce(
+          this._debouncer,
+          timeOut.after(500),
+          () => {
+            clearPageCallbacks();
+            comboBox.clearCache();
+            if (params.filter === '') {
+              // Fixes the case when the filter changes
+              // from '' to something else and back to ''
+              // within debounce timeout, and the
+              // DataCommunicator thinks it doesn't need to send data
+              comboBox.$server.resetDataCommunicator();
+            }
```

Comment:
Not sure if it's regression from these changes or not, but setting a filter, and then changing it back and forth quickly to the old filter leaves the combo-box in a loading state.
1. type `1` and wait for the data to load
2. type quickly any key + backspace so the filter is again `1`
3. notice that loading indicator keeps spinning and data is not visible

The server gets these requests:
0-50 filter: "1"
0-50 filter: "1"
Probably it thinks that no data needs to be sent because nothing has changes since the last call.

(not necessarily related to the highlighted lines, they seem to fix the same issue but only for empty filter)

Reply:
Not a regression, so doesn't block this PR. Probably the same issue as in #296, where I've added a comment.